server/service/transaction_service.rs

```diff
@@ -131,8 +133,10 @@ pub(crate) struct TransactionService {
     transaction: Option<Transaction>,
     request_queue: VecDeque<(Uuid, typeql::query::Pipeline)>,
     responders: HashMap<Uuid, (JoinHandle<()>, QueryStreamTransmitter)>,
-    running_write_query:
-        Option<(Uuid, JoinHandle<(Transaction, Result<(StreamQueryOutputDescriptor, Batch), Box<QueryError>>)>)>,
+    running_write_query: Option<(
+        Uuid,
+        JoinHandle<(Transaction, Result<Either<WriteQueryBatchAnswer, WriteQueryDocumentsAnswer>, Box<QueryError>>)>,
```

Comment:
I wanted to highlight that there is an `Either`, so I don't create a `type ... = Either<>`